README.md

```diff
@@ -98,4 +100,40 @@ We can extend the concept of the `as` syntax by generalizing the right hand side
+
+While an in band solution is more verbose, it is also more straightforward for developers to adopt. For smaller projects developers do not need to create an extra file by hand. For large project with many dependencies developers will not have to worry about creating a large manifest by compiling the metadata of all of their dependencies. Module authors will also not have to worry about shipping a manifest in order for consumers to be able to run their modules.
+
+The [import-maps proposal](https://github.com/WICG/import-maps) is a great example of an out of band manifest that could be created at installation time, but its focus is on modules and at this time all necessary information to generate an application's import map could be found in the package.json of its dependencies.
+
+### Are there cross-environment concerns?
+
+Absolutely. This proposal is not attempting to standardize the specific `type`s a host will be allowed to support. It will be up to each runtime to decide which `type`s to support, as well as how to handle the the case of an unsupported module `type` being imported.
+
+Module specifiers are an example of prior art for this type of decision making. It is left up to the host how to resolve a module specifier into a URL that will eventually be loaded.
+
+Independent of the potential cross-environment concerns it is not a problem that is specific to an in band solution. An out of band solution would also suffer from the risk of inconsistent implementation or support across runtimes.
+
+The topic of attribute divergence is further discussed in  [#34](https://github.com/tc39/proposal-module-attributes/issues/34).
+
+### Should the generalized constant form be supported in the first iteration?
+
+This proposal was initially created to serve a very specific purpose, unblocking security concerns raised about importing alternative module types (JSON, CSS, HTML). While there is no debate that the generalized constant form would be useful, it does pose the risk of increasing cross-environment concerns mentioned above.
+
+The champion group is very much in support of pursuing the generalized constant form in the long term, but would like to discuss with committee if it should be supported in the first iteration or not.
+
+### How would this proposal work with caching?
+
+Currently in [the ECMA262 spec](https://tc39.es/ecma262/#sec-hostresolveimportedmodule) passing the same module specifier and referrer there is guaranteed to get the same module. This proposal aims to have a similar guarantee; if a developer passes the same module specifier, referrer, and attributes there would be a guarantee of receiving the same module.
+
+It is possible that hosts will explore additional caching and coalescing beyond this.
```

Comment:
New wording:
"We expect that hosts will cache/coalesce further than this. For example, on the Web and similar environments, when providing a type module attribute, the module type would not form part of the cache key, where modules are always cached by the resolved URL. Instead, the type attribute is used simply for a check, which would cause the module graph to fail to load in the event of a mismatch."